From 30e5472936b2781d8c746a6f32ba8dc33a4ff88f Mon Sep 17 00:00:00 2001
From: kimsehun <se.hun.kim@intel.com>
Date: Fri, 12 Jul 2019 01:16:12 +0000
Subject: [PATCH] Add types_c.h and constants_c.h

Adding types_c.h on image_augmenter.h
and Adding constants_c.h on im2rec.cc
to catch CV_RGB2GRAY and other color change format

Signed-off-by: kimsehun <se.hun.kim@intel.com>
---
 src/io/image_augmenter.h | 1 +
 src/io/image_io.cc       | 1 +
 tools/im2rec.cc          | 2 ++
 3 files changed, 4 insertions(+)

diff --git a/src/io/image_augmenter.h b/src/io/image_augmenter.h
index e8a56ba2e..41811d623 100644
--- a/src/io/image_augmenter.h
+++ b/src/io/image_augmenter.h
@@ -34,6 +34,7 @@
 #include <string> // NOLINT(*)
 
 #include "../common/utils.h"
+#include <opencv2/imgproc/types_c.h>
 
 namespace mxnet {
 namespace io {
diff --git a/src/io/image_io.cc b/src/io/image_io.cc
index b3f7c40b2..581818e67 100644
--- a/src/io/image_io.cc
+++ b/src/io/image_io.cc
@@ -38,6 +38,7 @@
 #include <cstring>
 
 #include "../operator/elemwise_op_common.h"
+#include <opencv2/imgproc/types_c.h>
 
 #if MXNET_USE_OPENCV
   #include <opencv2/opencv.hpp>
diff --git a/tools/im2rec.cc b/tools/im2rec.cc
index 915b78029..8163d52a6 100644
--- a/tools/im2rec.cc
+++ b/tools/im2rec.cc
@@ -40,6 +40,8 @@
 #include <dmlc/recordio.h>
 #include <opencv2/opencv.hpp>
 #include "../src/io/image_recordio.h"
+#include <opencv2/imgcodecs/legacy/constants_c.h>
+#include <opencv2/imgproc/types_c.h>
 #include <random>
 /*!
  *\brief get interpolation method with given inter_method, 0-CV_INTER_NN 1-CV_INTER_LINEAR 2-CV_INTER_CUBIC
-- 
2.22.0

